Frontend/ui-customer/src/Flow.purs

```diff
@@ -2162,26 +2173,84 @@ checkAndUpdateSavedLocations state = do
 addLocationToRecents :: LocationListItemState -> HomeScreenState -> Boolean -> Boolean -> FlowBT String Unit
 addLocationToRecents item state srcServiceable destServiceable = do
   let serviceable = if (state.props.isSource == Just false) then destServiceable else srcServiceable
+      lat = if (state.props.isSource == Just false) then state.props.destinationLat else state.props.sourceLat
+      lon = if (state.props.isSource == Just false) then state.props.destinationLong else state.props.sourceLong
   case item.locationItemType of
     Just PREDICTION -> do
-        let lat = if (state.props.isSource == Just false) then state.props.destinationLat else state.props.sourceLat
-        let lon = if (state.props.isSource == Just false) then state.props.destinationLong else state.props.sourceLong
         saveToRecents item lat lon serviceable
-    _ -> saveToRecents item (fromMaybe 0.0 item.lat) (fromMaybe 0.0 item.lon) serviceable
+        when (state.props.isSource == Just false) $ do
+          setSuggestionsMapInLocal item lat lon serviceable state
+    _ -> do 
+        saveToRecents item (fromMaybe 0.0 item.lat) (fromMaybe 0.0 item.lon) serviceable
```

Comment:
if the lat lon is 0.0 it will save that as well in recents? can saveToRecents function have a check within it, if lat lon is 0.0 then not to save?

Reply:
in that case we can just make parameters of saveToRecents of Maybe type pass nothing in case item.lat is not there instead of passing 0.0

Reply:
Yes, Updated `saveToRecents`function  to only store the item when Lat, Lon is not 0.0